### ISSUE TITLE
Interactive Window: no stdout&stderr in interactive and notebook with logging and asyncio

### ISSUE THREAD

**Issue by NewUserHa** [microsoft/vscode-jupyter#5379]
## Environment data

-   VS Code version: 1.54.1
-   Jupyter Extension version (available under the Extensions sidebar): 2021.3.684299474
-   Python Extension version (available under the Extensions sidebar): 2021.3.680753044
-   OS (Windows | Mac | Linux distro) and version: win10 1803
-   Python and/or Anaconda version: 3.8.1
-   Type of virtual environment used: NA
-   Jupyter server running: Local

## Expected behaviour

At least, stdout will result.

## Actual behaviour

Output nothing.

but with ipython in CMD, both stdout and stderr will work.

isn't print() also write to stdout? then why logging to stdout can not work?

## Steps to reproduce:
![image](https://user-images.githubusercontent.com/32261870/113287379-17a72a80-9320-11eb-96a8-33d86f0193ce.png)
```python
import logging
import sys
from aiohttp import web

logging.basicConfig(level=logging.DEBUG, stream=sys.stdout # or stderr)

async def hello(request):
    return web.Response(text="Hello, world")

app = web.Application()
app.add_routes([web.get('/', hello)])
runner = web.AppRunner(app)
await runner.setup()
site = web.TCPSite(runner, 'localhost', 8080)
await site.start()
## use while await if in cmd, or app.run(...) instead
```
then open in browser: http://localhost:8080/

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
User belongs to experiment group 'pythonJoinMailingListVar2'
User belongs to experiment group 'jupyterTest'
User belongs to experiment group 'pythonDiscoveryModuleWithoutWatcher'
User belongs to experiment group 'pythonFindInterpreter'
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import notebook"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import notebook"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import jupyter"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import jupyter"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py pip list
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py pip list
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py jupyter kernelspec --version
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py jupyter kernelspec --version
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
Error 2021-04-01 19:28:22: Jupyter Kernel Spec not found for a local connection
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py pip list
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py pip list
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
Started kernel Python 3.8.1 64-bit
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py -c "import ipykernel"
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v
> C:\Program Files\Python38\python.exe c:\Users\user\.vscode\extensions\ms-toolsai.jupyter-2021.3.684299474\pythonFiles\pyvsc-run-isolated.py vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_launcher_daemon -v

```

</p>
</details>

**Comment by rchiodo** [comment]
Thanks for the issue.

Jupyter classic seems to behave the same way. If you attempt this same code in Jupyter does it print to stdout?

**Comment by NewUserHa** [comment]
only have ipython.
so for now, can't test what behavior of jupyter is.

**Comment by DonJayamanne** [comment]
I can confirm this is a bug in our interactive window.
Output is not being displayed. The output from the logger is displayed in Native Notebooks

**Comment by NewUserHa** [comment]
both `stdout` and `stderr` are displayed in Native Notebooks cell or its server logging? or just `stdout`?

**Comment by DonJayamanne** [comment]
I didn't check which ones, but we're definitely getting output in nativenotebooks & not getting anything in interactive window.
Whether its stdout/stderr can be looked at when we're fixing this issue.

**Comment by NewUserHa** [comment]
thanks!
wish both can be displayed by config.

**Comment by lfriedri** [comment]
I think I can see the same issue in a different usecase:

### Environment
VS Code version: 1.58.2
Jupyter Extension version (available under the Extensions sidebar): 2021.8.1054968649
Python Extension version (available under the Extensions sidebar): 2021.7.1060902895
OS (Windows | Mac | Linux distro) and version: win10
Python and/or Anaconda version: 3.8.6
Type of virtual environment used: NA
Jupyter server running: Local

### Steps to reproduce
Run the following in interactive window (put the %gui wx into interactive window manually at first)
```
#%gui wx

import wx

class MyFrame(wx.Frame):
    def __init__(self):
        wx.Frame.__init__(self, None)
        
        self.btn = wx.Button(self)
        self.btn.Bind(wx.EVT_BUTTON, self.onButton)

        self.Layout()
        self.Show()
    
    def onButton(self, evt):
        print('button')

f = MyFrame()
```

### Expected behaviour
When the button is pressed, print output appears in interactive window.

### Actual behaviour
Frame is displayed, but no print output.

### Note
With "Jupyter Create New Blank Notebook" the behaviour is as expected.

**Comment by DonJayamanne** [comment]
Thanks for filing this issue, we're currently aware of this issue and are planning on resolving this soon.
